pandas/_libs/tslib.pyx

```diff
@@ -140,20 +140,44 @@ def format_array_from_datetime(
     if na_rep is None:
         na_rep = 'NaT'
 
-    # if we don't have a format nor tz, then choose
-    # a format based on precision
-    basic_format = format is None and tz is None
-    if basic_format:
-        reso_obj = get_resolution(values, reso=reso)
-        show_ns = reso_obj == Resolution.RESO_NS
-        show_us = reso_obj == Resolution.RESO_US
-        show_ms = reso_obj == Resolution.RESO_MS
+    if tz is None:
+        # if we don't have a format nor tz, then choose
+        # a format based on precision
+        basic_format = format is None
+        if basic_format:
+            reso_obj = get_resolution(values, reso=reso)
+            show_ns = reso_obj == Resolution.RESO_NS
+            show_us = reso_obj == Resolution.RESO_US
+            show_ms = reso_obj == Resolution.RESO_MS
+
+        elif format == "%Y-%m-%d %H:%M:%S":
```

Comment:
(ref my general comment) Here and below, should I consider `reso` ? My guess is not, since the explicit presence of `format` overrides any kind of guessing. However what happens if format is explicitly set as "%Y-%m-%d %H:%M:%S.%f" (microsecond) and `reso` is explicitly set as milliseconds ? Who wins ?

Reply:
correct, if format is given then you can ignore reso.  Once we've used it to get show_ns, show_us, show_ms, you shouldn't need to worry about it anymore

Reply:
perfect thanks !